code/lib/codemod/src/transforms/__tests__/mdx-to-csf.test.ts

```diff
@@ -0,0 +1,412 @@
+  const input = dedent`
+      import { Meta } from '@storybook/addon-docs';
+
+      export const args = { bla: 1 };
+      
+      <Meta title="foobar" args={{...args}} parameters={{a: '1'}} />
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      const args = { bla: 1 };
+
+      export default {
+        title: 'foobar',
+
+        args: {
+          ...args,
+        },
+
+        parameters: {
+          a: '1',
+        },
+      };
+
+    `);
+});
+
+test('extract all story attributes', () => {
+  const input = dedent`
+      import { Meta, Story } from '@storybook/addon-docs';
+
+      export const args = { bla: 1 };
+      
+      <Meta title="foobar" />
+
+      <Story name="foo">bar</Story>
+      
+      <Story 
+        name="Unchecked"
+        args={{
+          ...args,
+          label: 'Unchecked',
+        }}>    
+        {Template.bind({})}
+      </Story>
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      const args = { bla: 1 };
+
+      export default {
+        title: 'foobar',
+      };
+
+      export const Foo = {
+        render: () => 'bar',
+        name: 'foo',
+      };
+
+      export const Unchecked = {
+        render: Template.bind({}),
+        name: 'Unchecked',
+
+        args: {
+          ...args,
+          label: 'Unchecked',
+        },
+      };
+
+    `);
+});
+
+test('story child is jsx', () => {
+  const input = dedent`
+      import { Canvas, Meta, Story } from '@storybook/addon-docs';
+      import { Button } from './button';
+      
+      <Story name="Primary">
+        <Button>
+          <div>Hello!</div>
+        </Button>
+      </Story>
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      import { Button } from './button';
+      export default {};
+
+      export const Primary = {
+        render: () => (
+          <Button>
+            <div>Hello!</div>
+          </Button>
+        ),
+
+        name: 'Primary',
+      };
+
+    `);
+});
+
+test('story child is arrow function', () => {
+  const input = dedent`
+      import { Canvas, Meta, Story } from '@storybook/addon-docs';
+      import { Button } from './button';
+      
+      <Story name="Primary">
+        {(args) => <Button />}
+      </Story>
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      import { Button } from './button';
+      export default {};
+
+      export const Primary = {
+        render: (args) => <Button />,
+        name: 'Primary',
+      };
+
+    `);
+});
+
+test('story child is identifier', () => {
```

Comment:
Can we have a test for "story child is CSF3"?

Reply:
Added it